File: scala/src/main/scala/io/swagger/client/model/VehicleType.scala
package io.swagger.client.model




case class VehicleType (
  /* Unique identifier for the vehicle type */
  type_id: String,
  /* Profile of vehicle type */
  profile: String,
  /* array of capacity dimensions */
  capacity: List[Integer],
  /* speed_factor of vehicle type */
  speed_factor: Double,
  /* service time factor of vehicle type */
  service_time_factor: Double)
  

File: dart/lib/model/vehicle_type.dart
part of swagger.api;


@Entity()
class VehicleType {
  /* Unique identifier for the vehicle type */
  @Property(name: 'type_id')
  String typeId = null;
  
/* Profile of vehicle type */
  @Property(name: 'profile')
  String profile = null;
  //enum profileEnum {  car,  bike,  foot,  mtb,  motorcycle,  racingbike,  truck,  small_truck,  bus,  };
/* array of capacity dimensions */
  @Property(name: 'capacity')
  List<int> capacity = [];
  
/* speed_factor of vehicle type */
  @Property(name: 'speed_factor')
  double speedFactor = null;
  
/* service time factor of vehicle type */
  @Property(name: 'service_time_factor')
  double serviceTimeFactor = null;
  
  VehicleType();

  @override
  String toString()  {
    return 'VehicleType[typeId=$typeId, profile=$profile, capacity=$capacity, speedFactor=$speedFactor, serviceTimeFactor=$serviceTimeFactor, ]';
  }

}

